lib/SimpleHttpClient/SslClientConnection.cpp

```diff
@@ -302,9 +307,30 @@ bool SslClientConnection::connectSocket() {
 
   _errorDetails.clear();
   _socket = _endpoint->connect(_connectTimeout, _requestTimeout);
+  if (_isSocketNonBlocking) {
+    _socketFlags = fcntl(_socket.fileDescriptor, F_GETFL, 0);
+    if (_socketFlags == -1) {
+      _errorDetails = "Socket file descriptor read returned with error " +
+                      std::to_string(errno);
+      _isConnected = false;
+      return false;
+    }
+    if (fcntl(_socket.fileDescriptor, F_SETFL, _socketFlags | O_NONBLOCK) ==
+        -1) {
+      _errorDetails = "Attempt to create non-blocking socket generated error " +
+                      std::to_string(errno);
+      _isConnected = false;
+      return false;
+    }
+  }
```

Comment:
All this code should only be executed after the check for `TRI_isvalidsocket(_socket)`. Because why should we execute it on a socket that we already does not work?